pkg/pool-linear/test/LinearPool.test.ts

```diff
@@ -662,222 +658,4 @@ describe('LinearPool', function () {
-    let timestamp: BigNumber;
-    let wrappedTokenRateProvider: Contract;
-    const wrappedTokenRateCacheDuration = MINUTE * 20;
-
-    const scaleRate = (rate: BigNumber) => rate.mul(bn(10).pow(18 - wrappedToken.decimals));
-
-    sharedBeforeEach('deploy pool', async () => {
-      wrappedTokenRateProvider = await deploy('v2-pool-utils/MockRateProvider');
-      timestamp = await currentTimestamp();
-      await deployPool({ mainToken, wrappedToken, wrappedTokenRateProvider, wrappedTokenRateCacheDuration });
-    });
-
-    it('initializes correctly', async () => {
-      const provider = await pool.getWrappedTokenRateProvider();
-      expect(provider).to.be.equal(wrappedTokenRateProvider.address);
-
-      const { rate, duration, expires } = await pool.getWrappedTokenRateCache();
-      expect(rate).to.be.equal(fp(1));
-      expect(duration).to.be.equal(wrappedTokenRateCacheDuration);
-      expect(expires).to.be.at.least(timestamp.add(wrappedTokenRateCacheDuration));
-    });
-
-    describe('scaling factors', () => {
-      const itAdaptsTheScalingFactorsCorrectly = () => {
-        const expectedBptScalingFactor = fp(1);
-        const expectedMainTokenScalingFactor = fp(1);
-
-        it('adapt the scaling factors with the price rate', async () => {
-          const scalingFactors = await pool.getScalingFactors();
-
-          const expectedWrappedTokenScalingFactor = scaleRate(await wrappedTokenRateProvider.getRate());
-          expect(scalingFactors[pool.wrappedIndex]).to.be.equal(expectedWrappedTokenScalingFactor);
-          expect(await pool.getScalingFactor(wrappedToken)).to.be.equal(expectedWrappedTokenScalingFactor);
-
-          expect(scalingFactors[pool.mainIndex]).to.be.equal(expectedMainTokenScalingFactor);
-          expect(await pool.getScalingFactor(mainToken)).to.be.equal(expectedMainTokenScalingFactor);
-
-          expect(scalingFactors[pool.bptIndex]).to.be.equal(expectedBptScalingFactor);
-          expect(await pool.getScalingFactor(pool.bptToken)).to.be.equal(expectedBptScalingFactor);
-        });
-      };
-
-      context('with a price rate above 1', () => {
-        sharedBeforeEach('mock rate', async () => {
-          await wrappedTokenRateProvider.mockRate(fp(1.1));
-          await pool.updateWrappedTokenRateCache();
-        });
-
-        itAdaptsTheScalingFactorsCorrectly();
-      });
-
-      context('with a price rate equal to 1', () => {
-        sharedBeforeEach('mock rate', async () => {
-          await wrappedTokenRateProvider.mockRate(fp(1));
-          await pool.updateWrappedTokenRateCache();
-        });
-
-        itAdaptsTheScalingFactorsCorrectly();
-      });
-
-      context('with a price rate below 1', () => {
-        sharedBeforeEach('mock rate', async () => {
-          await wrappedTokenRateProvider.mockRate(fp(0.99));
-          await pool.updateWrappedTokenRateCache();
-        });
-
-        itAdaptsTheScalingFactorsCorrectly();
-      });
-    });
```

Comment:
These tests are important: they test the rate is correctly applied to the scaling factors. Let's add a simple setter to the `MockLinearPool` for the rate, and keep these tests.